src/main/java/seedu/commands/BudgetCommand.java

```diff
@@ -104,14 +107,18 @@ private void addBudget(SpendingList spendings, double budget, String timeRange)
         String formatedBudget = Ui.formatPrintDouble(budget);
         switch (timeRange) {
         case DAILY:
+            checkUserBudgetEntered(budget, spendings.getMonthlyBudget(), "Daily budget", "monthly budget");
             spendings.setDailyBudget(budget);
             Ui.printWithTab(DAILY_BUDGET_SUCCESS_MESSAGE + formatedBudget);
             break;
         case MONTHLY:
+            checkUserBudgetEntered(budget, spendings.getYearlyBudget(), "Monthly budget", "yearly budget");
+            checkUserBudgetEntered(spendings.getDailyBudget(), budget, "Daily budget", "monthly budget");
             spendings.setMonthlyBudget(budget);
             Ui.printWithTab(MONTHLY_BUDGET_SUCCESS_MESSAGE + formatedBudget);
             break;
         case YEARLY:
+            checkUserBudgetEntered(spendings.getMonthlyBudget(), budget, "Monthly budget", "yearly budget");
```

Comment:
maybe should refractor the magic strings out as they are used multiple times

Reply:
done